src/functions/module_lookup.ts

```diff
@@ -127,6 +131,53 @@ export const HLOOKUP: AddFunctionDescription = {
   isExported: true,
 };
 
+// -----------------------------------------------------------------------------
+// INDEX
+// -----------------------------------------------------------------------------
+
+export const INDEX: AddFunctionDescription = {
+  description: _lt(`Returns the content of a cell, specified by row and column offset.`),
+  args: args(`
+      reference (any, range) ${_lt("The range of cells from which the value is returned.")}
+      row (number) ${_lt(
+        "The index of the row to be returned from within the reference range of cells."
+      )}
+      column (number) ${_lt(
+        "The index of the column to be returned from within the reference range of cells."
+      )}
+`),
+  returns: ["ANY"],
+  computeFormat: (reference: Arg, row: PrimitiveArg, column: PrimitiveArg) => {
+    const _row = toNumber(row.value);
+    const _column = toNumber(column.value);
+    return reference[_column - 1][_row - 1]?.format;
+  },
+  compute: function (reference: ArgValue, row: PrimitiveArgValue, column: PrimitiveArgValue): any {
+    const _reference = (isMatrix(reference) ? reference : [[reference]]) as ArgValue[][];
```

Comment:
If you copied the typing of `isMatrix` correctly you wouldn't have to cast the result (your cast is wrong btw).

```suggestion
    const _reference = isMatrix(reference) ? reference : [[reference]]);
    
export function isMatrix(x: any): x is any[][] {
  return Array.isArray(x) && Array.isArray(x[0]);
}

```